gcc/d/decl.cc

```diff
@@ -1615,8 +1607,6 @@ finish_thunk (tree thunk, tree function)
 
       if (!stdarg_p (TREE_TYPE (thunk)))
 	{
-	  /* Put generic thunk into COMDAT.  */
-	  d_comdat_linkage (thunk);
```

Comment:
I don't think this is needed, as `DECL_WEAK`, `DECL_COMDAT`, et al. are copied from function to thunk earlier.